app/services/UserServiceImpl.js

```diff
@@ -29,7 +29,28 @@ module.exports = function(userDao) {
     });
   };
 
+  const findByUsername = function(username, password) {
```

Comment:
`findByUsername` isn't really accurate because this does more than simply `findByUsername`. The method name should better convey what the method does so something like `authenticateByUsername` would be more accurate.